keras_nlp/tokenizers/byte_tokenizer_test.py

```diff
@@ -0,0 +1,219 @@
+                    150,
+                    129,
+                    226,
+                    150,
+                    130,
+                    226,
+                    150,
+                    131,
+                    226,
+                    150,
+                    132,
+                ],
+            ]
+        )
+
+        tokenizer = ByteTokenizer()
+        detokenize_output = tokenizer.detokenize(input_data)
+        self.assertAllEqual(detokenize_output, ["hello", "fun", "▀▁▂▃▄"])
+
+    def test_detokenize_replace_error(self):
+        input_data = tf.ragged.constant([[104, 101, 226, 150, 108, 108, 111]])
+
+        tokenizer = ByteTokenizer(errors="replace", replacement_char=341)
+        detokenize_output = tokenizer.detokenize(input_data)
+        self.assertAllEqual(detokenize_output, [b"he\xc5\x95llo"])
+
+    def test_detokenize_ignore_error(self):
+        input_data = tf.ragged.constant([[104, 101, 226, 150, 108, 108, 111]])
+
+        tokenizer = ByteTokenizer(errors="ignore")
+        detokenize_output = tokenizer.detokenize(input_data)
+        self.assertAllEqual(detokenize_output, [b"hello"])
+
+    def test_detokenize_strict_error(self):
+        input_data = tf.ragged.constant([[104, 101, 226, 150, 108, 108, 111]])
+
+        tokenizer = ByteTokenizer(errors="strict")
+        with self.assertRaises(tf.errors.InvalidArgumentError):
+            _ = tokenizer.detokenize(input_data)
+
+    def test_accessors(self):
+        tokenizer = ByteTokenizer()
+        self.assertEqual(tokenizer.vocabulary_size(), 256)
+        self.assertEqual(
+            tokenizer.get_vocabulary(),
+            [i.tobytes() for i in np.arange(256, dtype=np.uint8)],
+        )
+        self.assertEqual(tokenizer.id_to_token(9), b"\t")
+        self.assertEqual(tokenizer.id_to_token(123), b"{")
+        self.assertEqual(tokenizer.token_to_id(b"\t"), 9)
+        self.assertEqual(tokenizer.token_to_id(b"{"), 123)
+
+    def test_lowercase(self):
+        input_data = ["HeLlO wOrLd"]
+        tokenizer = ByteTokenizer()
+        call_output = tokenizer(input_data)
+        self.assertAllEqual(
+            call_output,
+            [[104, 101, 108, 108, 111, 32, 119, 111, 114, 108, 100]],
+        )
+
+    def test_skip_lowercase(self):
+        input_data = ["HeLlO wOrLd"]
+        tokenizer = ByteTokenizer(lowercase=False)
+        call_output = tokenizer(input_data)
+        self.assertAllEqual(
+            call_output, [[72, 101, 76, 108, 79, 32, 119, 79, 114, 76, 100]]
+        )
+
+    def test_functional_model(self):
+        input_data = tf.constant(["hello", "fun", "▀▁▂▃▄"])
+        tokenizer = ByteTokenizer()
+        inputs = keras.Input(dtype="string", shape=())
+        outputs = tokenizer.detokenize(tokenizer.tokenize(inputs))
+        model = keras.Model(inputs, outputs)
+        model_output = model(input_data)
+        self.assertAllEqual(model_output, ["hello", "fun", "▀▁▂▃▄"])
+
+    def test_config_1(self):
```

Comment:
come up with a more descriptive name than _1 and _2 here, or fold into one test.